core/src/core/block.rs

```diff
@@ -660,6 +662,30 @@ impl Block {
 		Ok(())
 	}
 
+	// TODO - how do we verify Merkle Proof here?
```

Comment:
Sounds like you figured this out? Maybe explain move up the comment below that explains what's verified here vs what will be verified later?